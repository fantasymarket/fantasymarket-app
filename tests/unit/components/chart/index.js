import React from 'react';
import { render } from '@testing-library/react';

import StockChart from '@components/chart/stock';
import StockChartMinimal from '@components/chart/stock-minimal';

describe('StockChart component', () => {
	const container = render(<StockChart />);

	it('renders the component', () => {
		expect(container.baseElement).toMatchSnapshot();
	});
});

describe('StockChartMinimal component', () => {
	const container = render(<StockChartMinimal />);

	it('renders the component', () => {
		expect(container.baseElement).toMatchSnapshot();
	});
});
