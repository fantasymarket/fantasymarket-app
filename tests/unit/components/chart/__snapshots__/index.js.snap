// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockChart component renders the component 1`] = `
<body>
  <div>
    <div
      class="sc-AxjAm bntWza"
    >
      <div />
    </div>
  </div>
  <div>
    <div
      class="sc-AxirZ gMSJvy"
      title="Full Chart"
    >
      <div
        class="VictoryContainer"
        style="user-select: none; pointer-events: none; position: relative; width: 600px; height: 350px;"
      >
        <svg
          aria-labelledby="victory-container-1-title"
          height="350"
          role="img"
          style="pointer-events: all; width: 600px; height: 350px;"
          width="600"
        >
          <title
            id="victory-container-1-title"
          >
            Chart
          </title>
          <g
            clip-path="url(#victory-clip-2)"
            style="height: 100%; width: 100%; user-select: none;"
          >
            <defs>
              <clippath
                id="victory-clip-2"
              >
                <rect
                  height="250"
                  vector-effect="non-scaling-stroke"
                  width="500"
                  x="50"
                  y="50"
                />
              </clippath>
            </defs>
            <path
              d="M50,300L112.5,224.63235294117644L206.22861054072553,187.86764705882356L268.7286105407255,237.5L284.3429158110883,261.39705882352945L331.2286105407255,277.94117647058823L346.8429158110883,255.88235294117644L409.3429158110883,231.98529411764707L456.2286105407255,197.05882352941177L471.8429158110883,134.55882352941177L518.7286105407255,106.98529411764704L550,50"
              role="presentation"
              shape-rendering="auto"
              style="fill: transparent; stroke: #00f100; stroke-width: 2;"
            />
          </g>
          <g
            role="presentation"
          >
            <line
              role="presentation"
              shape-rendering="auto"
              style="stroke: #252525; fill: transparent; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
              vector-effect="non-scaling-stroke"
              x1="50"
              x2="550"
              y1="300"
              y2="300"
            />
            <g
              role="presentation"
            >
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-1-tickLabels-0"
                x="111.17385352498289"
                y="322.97"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                  text-anchor="middle"
                  x="111.17385352498289"
                >
                  2005
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-1-tickLabels-1"
                x="189.28815879534565"
                y="322.97"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                  text-anchor="middle"
                  x="189.28815879534565"
                >
                  2010
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-1-tickLabels-2"
                x="267.40246406570844"
                y="322.97"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                  text-anchor="middle"
                  x="267.40246406570844"
                >
                  2015
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-1-tickLabels-3"
                x="345.5167693360712"
                y="322.97"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                  text-anchor="middle"
                  x="345.5167693360712"
                >
                  2020
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-1-tickLabels-4"
                x="423.6738535249829"
                y="322.97"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                  text-anchor="middle"
                  x="423.6738535249829"
                >
                  2025
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-1-tickLabels-5"
                x="501.7881587953457"
                y="322.97"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                  text-anchor="middle"
                  x="501.7881587953457"
                >
                  2030
                </tspan>
              </text>
            </g>
          </g>
          <g
            role="presentation"
          >
            <line
              role="presentation"
              shape-rendering="auto"
              style="stroke: #252525; fill: transparent; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
              vector-effect="non-scaling-stroke"
              x1="50"
              x2="50"
              y1="50"
              y2="300"
            />
            <g
              role="presentation"
            >
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-2-tickLabels-0"
                x="39"
                y="277.3964705882353"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                  text-anchor="end"
                  x="39"
                >
                  380
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-2-tickLabels-1"
                x="39"
                y="240.63176470588238"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                  text-anchor="end"
                  x="39"
                >
                  400
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-2-tickLabels-2"
                x="39"
                y="203.86705882352942"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                  text-anchor="end"
                  x="39"
                >
                  420
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-2-tickLabels-3"
                x="39"
                y="167.10235294117643"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                  text-anchor="end"
                  x="39"
                >
                  440
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-2-tickLabels-4"
                x="39"
                y="130.33764705882353"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                  text-anchor="end"
                  x="39"
                >
                  460
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-2-tickLabels-5"
                x="39"
                y="93.5729411764706"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                  text-anchor="end"
                  x="39"
                >
                  480
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-2-tickLabels-6"
                x="39"
                y="56.80823529411764"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                  text-anchor="end"
                  x="39"
                >
                  500
                </tspan>
              </text>
            </g>
          </g>
        </svg>
        <div
          style="z-index: 99; position: absolute; top: 0px; left: 0px; width: 600px; height: 350px;"
        >
          <svg
            height="350"
            style="overflow: visible; width: 600px; height: 350px;"
            width="600"
          />
        </div>
      </div>
      <div
        class="VictoryContainer"
        style="user-select: none; pointer-events: none; position: relative; width: 600px; height: 90px;"
      >
        <svg
          aria-labelledby="victory-container-3-title"
          height="90"
          role="img"
          style="pointer-events: all; width: 600px; height: 90px;"
          width="600"
        >
          <title
            id="victory-container-3-title"
          >
            Chart Zoom Selector
          </title>
          <g
            role="presentation"
          >
            <line
              role="presentation"
              shape-rendering="auto"
              style="stroke: #252525; color: white; fill: transparent; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
              vector-effect="non-scaling-stroke"
              x1="50"
              x2="550"
              y1="60"
              y2="60"
            />
            <g
              role="presentation"
            >
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-0-tickLabels-0"
                x="50"
                y="82.97"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                  text-anchor="middle"
                  x="50"
                >
                  2000
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-0-tickLabels-1"
                x="121.45650813516896"
                y="82.97"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                  text-anchor="middle"
                  x="121.45650813516896"
                >
                  2005
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-0-tickLabels-2"
                x="192.8739048811014"
                y="82.97"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                  text-anchor="middle"
                  x="192.8739048811014"
                >
                  2010
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-0-tickLabels-3"
                x="264.2913016270338"
                y="82.97"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                  text-anchor="middle"
                  x="264.2913016270338"
                >
                  2015
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-0-tickLabels-4"
                x="335.7086983729662"
                y="82.97"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                  text-anchor="middle"
                  x="335.7086983729662"
                >
                  2020
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-0-tickLabels-5"
                x="407.16520650813516"
                y="82.97"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                  text-anchor="middle"
                  x="407.16520650813516"
                >
                  2025
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-0-tickLabels-6"
                x="478.5826032540676"
                y="82.97"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                  text-anchor="middle"
                  x="478.5826032540676"
                >
                  2030
                </tspan>
              </text>
            </g>
            <g
              role="presentation"
            >
              <text
                direction="inherit"
                dx="0"
                id="chart-axis-0-tickLabels-7"
                x="550"
                y="82.97"
              >
                <tspan
                  dx="0"
                  dy="0"
                  style="font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
                  text-anchor="middle"
                  x="550"
                >
                  2035
                </tspan>
              </text>
            </g>
          </g>
          <g
            clip-path="url(#victory-clip-4)"
            style="height: 100%; width: 100%; user-select: none;"
          >
            <defs>
              <clippath
                id="victory-clip-4"
              >
                <rect
                  height="60"
                  vector-effect="non-scaling-stroke"
                  width="500"
                  x="50"
                  y="0"
                />
              </clippath>
            </defs>
            <path
              d="M64.31476846057572,60L121.45650813516896,41.91176470588235L207.1495619524406,33.08823529411765L264.2913016270338,45L278.56695869837296,50.73529411764706L321.433041301627,54.705882352941174L335.7086983729662,49.41176470588235L392.8504380475594,43.6764705882353L435.7165206508135,35.294117647058826L449.9921777221527,20.294117647058826L492.8582603254068,13.676470588235292L521.4486858573216,0"
              role="presentation"
              shape-rendering="auto"
              style="fill: transparent; stroke: #00f100; stroke-width: 2;"
            />
          </g>
          <rect
            cursor="move"
            height="60"
            style="stroke: transparent; fill: black; fill-opacity: 0.1;"
            vector-effect="non-scaling-stroke"
            width="500"
            x="50"
            y="0"
          />
          <rect
            cursor="ew-resize"
            height="60"
            style="stroke: transparent; fill: transparent;"
            vector-effect="non-scaling-stroke"
            width="8"
            x="46"
            y="0"
          />
          <rect
            cursor="ew-resize"
            height="60"
            style="stroke: transparent; fill: transparent;"
            vector-effect="non-scaling-stroke"
            width="8"
            x="546"
            y="0"
          />
        </svg>
        <div
          style="z-index: 99; position: absolute; top: 0px; left: 0px; width: 600px; height: 90px;"
        >
          <svg
            height="90"
            style="overflow: visible; width: 600px; height: 90px;"
            width="600"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`StockChartMinimal component renders the component 1`] = `<body />`;
